### PR TITLE
Add `mock_build` to allow for faster circuit building and partial witness generation 

### PR DESCRIPTION
**Pull request by puma314**

### CODE REVIEW

File: plonky2/src/plonk/circuit_builder.rs [hunk 1]
```diff
@@ -1023,15 +1026,18 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
...
-        let constants_sigmas_vecs = [constant_vecs, sigma_vecs.clone()].concat();
-        let constants_sigmas_commitment = PolynomialBatch::<F, C, D>::from_values(
-            constants_sigmas_vecs,
-            rate_bits,
-            PlonkOracle::CONSTANTS_SIGMAS.blinding,
-            cap_height,
-            &mut timing,
-            Some(&fft_root_table),
-        );
+        let mut constants_sigmas_commitment = PolynomialBatch::<F, C, D>::default();
```
Comment (npwardberkeley) [inline plonky2/src/plonk/circuit_builder.rs:1029]: nit: I think
`let constants_sigmas_commitment = if commit_to_sigma {
    ...
} else {
    PolynomialBatch::<F, C, D>::default()
}` would be cleaner

File: plonky2/src/plonk/circuit_data.rs [hunk 2]
```diff
@@ -111,6 +111,22 @@ impl CircuitConfig {
     }
 }
 
+/// Circuit data required by the prover or the verifier.
```
Comment (npwardberkeley) [inline plonky2/src/plonk/circuit_data.rs:114]: Update the comment?

File: plonky2/src/plonk/circuit_data.rs [hunk 3]
```diff
@@ -111,6 +111,22 @@ impl CircuitConfig {
...
+#[derive(Eq, PartialEq, Debug)]
+pub struct MockCircuitData<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize>
+{
+    pub prover_only: ProverOnlyCircuitData<F, C, D>,
+    pub common: CommonCircuitData<F, D>,
+}
+
+impl<'a, F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize>
```
Comment (npwardberkeley) [inline plonky2/src/plonk/circuit_data.rs:122]: Fix clippy error please!

**Comment by puma314** [comment]
Thanks for the review @npwardberkeley, fixed all the comments--PTAL.

**Pull request merged**